src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -508,7 +512,7 @@ public void leaveToken(DetailAST ast) {
             else if (TokenUtil.isOfType(type, ASSIGNMENTS)) {
                 assignDepth--;
             }
-            else if (isSurrounded(ast) && checkAroundOperators(ast)) {
+            else if (isSurrounded(ast) && unnecessaryParenAroundOperators(ast)) {
```

Comment:
Condition is more clear to read, is surrounded, and has unnecessary parens then log an error message.